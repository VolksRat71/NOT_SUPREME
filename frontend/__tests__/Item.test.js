import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const mockItem = {
    id: 'ABC123',
    title: 'User Item',
    description: 'Mock item description',
    price: 123534,
    image: 'mockItem.png',
    largeImage: 'mockLargeItem.png'
};

describe('<Item />', () => {

    it('Renders and matched the snapshot ', () => {
        const wrapper = shallow(<ItemComponent item={mockItem} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })


    //! Commented out, its too ridged for our application
    // const wrapper = shallow(<ItemComponent item={mockItem} />);

    // const elementParse = (element, item) => {
    //     switch (item) {
    //         case 'text':
    //             return wrapper.find(element).children().text()
    //             break;
    //         case 'filter':
    //             return wrapper.find(element).children()
    //             break;
    //         case 'attr':
    //             return wrapper.find(element).props()
    //             break;
    //         default:
    //             return wrapper.find(element).find(item)
    //     }
    // };

    // it('Renders image properly ', () => {
    //     expect(elementParse('img', 'attr').src).toBe(mockItem.image);
    //     expect(elementParse('img', 'attr').alt).toBe(mockItem.title);
    // });

    // it('Renders price and title properly ', () => {
    //     expect(elementParse('PriceTag', 'text')).toBe('$50');
    //     expect(elementParse('Title a', 'text')).toBe(mockItem.title);
    // });

    // it('Renders buttons properly ', () => {
    //     expect(elementParse('.buttonList', 'filter')).toHaveLength(3);
    //     expect(elementParse('.buttonList', 'Link')).toHaveLength(1);
    //     expect(elementParse('.buttonList', 'AddToCart').exists()).toBe(true)
    //     expect(elementParse('.buttonList', 'DeleteItem').exists()).toBe(true)
    // });
})
