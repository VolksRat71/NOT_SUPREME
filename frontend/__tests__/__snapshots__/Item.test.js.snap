// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Renders and matched the snapshot  1`] = `
<ItemStyles__Item>
  <img
    alt="User Item"
    src="mockItem.png"
  />
  <Link
    href={
      Object {
        "pathname": "/item",
        "query": Object {
          "id": "ABC123",
        },
      }
    }
  >
    <Title>
      <a>
        User Item
      </a>
    </Title>
  </Link>
  <PriceTag>
    $1,235.34
  </PriceTag>
  <p>
    Mock item description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <button>
        Edit
      </button>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
